src/datasets/graphs/temporalbrains.jl

```diff
@@ -0,0 +1,81 @@
+function __init__temporalbrains()
+    DEPNAME = "TemporalBrains"
+    LINK = "http://www-sop.inria.fr/members/Aurora.Rossi/index.html"
+    register(ManualDataDep(DEPNAME,
+                           """
+                           Dataset: $DEPNAME
+                           Website : $LINK
+                           """))
+end
+
+
+function tb_datadir(dir = nothing)
+    dir = isnothing(dir) ? datadep"TemporalBrains" : dir
+    LINK = "http://www-sop.inria.fr/members/Aurora.Rossi/data/LabelledTBN.zip"
+    if length(readdir((dir))) == 0
+        DataDeps.fetch_default(LINK, dir)
+        currdir = pwd()
+        cd(dir) # Needed since `unpack` extracts in working dir
+        DataDeps.unpack(joinpath(dir, "LabelledTBN.zip"))
+        # conditions when unzipped folder is our required data dir
+        cd(currdir)
+    end
+    @assert isdir(dir)
+    return dir
+end
+
+
+function create_tbdataset(dir, thre)
+    name_filelabels = joinpath(dir, "LabelledTBN", "labels.txt")
+    filelabels = open(name_filelabels, "r")
+    temporalgraphs = Vector{MLDatasets.TemporalSnapshotsGraph}(undef, 1000)
+
+    for (i,line) in enumerate(eachline(filelabels))
+        id, gender, age = split(line)
+        name_network_file = joinpath(dir, "LabelledTBN", "networks", id * "_ws60_wo30_tuk0_pearson_schaefer_100.txt")
+        
+        data = readdlm(name_network_file,',',Float32; skipstart = 1) 
+    
+        data_thre = view(data,view(data,:,4) .> thre,:)
+        data_thre_int = Int.(view(data_thre,:,1:3))
+
+        activation = [zeros(Float32, 102) for _ in 1:27]
+        for t in 1:27
+             for n in 1:102
+                rows = (view(data_thre_int,:,1).==n .&& view(data_thre_int,:,3).==t)
+                activation[t][n] = mean(view(data_thre,rows,4))
+             end
+        end
+
+        temporalgraphs[i] = TemporalSnapshotsGraph(num_nodes=ones(Int,27)*102, edge_index = (data_thre_int[:,1], data_thre_int[:,2], data_thre_int[:,3]), node_data= activation, graph_data= (g = gender, a = age))
+    end
+    return temporalgraphs
+end
+
+"""
+    TemporalBrains(; dir=nothing, threshold_value = 0.6)
+
+The TemporalBrains dataset contains a collection of temporal brain networks of 1000 subjects obtained from resting-state fMRI data from the Human Connectome Project (HCP).
+
+The number of nodes is fixed for each of the 27 snapshots at 102, instead the edges change over time.
+    
+Each node feature, representing its average activation at each snapshot, is contained in `node_data`.
```

Comment:
This is not cleat to me. The average activation in the window preceding the snapshot?